lib/CachePlugin.js

```diff
@@ -6,18 +6,39 @@
 
 const asyncLib = require("neo-async");
 
+/** @typedef {import("./Compiler.js")} Compiler */
+
+/**
+ * Adds a cache to the compiler, where modules are cached.
+You can pass a cache object, where the modules are cached. Otherwise one is created per plugin instance.
+ */
 class CachePlugin {
+	/**
+	 *
+	 * @param {Object} [cache] cache object, where the modules are cached
+	 */
 	constructor(cache) {
 		this.cache = cache || {};
 		this.FS_ACCURACY = 2000;
 	}
 
+	/**
+	 *
+	 * @param {Compiler} compiler webpack compiler
+	 * @returns {void}
+	 */
 	apply(compiler) {
 		if (Array.isArray(compiler.compilers)) {
 			compiler.compilers.forEach((c, idx) => {
 				new CachePlugin((this.cache[idx] = this.cache[idx] || {})).apply(c);
 			});
 		} else {
+			/**
+			 *
+			 * @param {Compiler} compiler webpack compiler
+			 * @param {Object} cache cache object, where the modules are cached
```

Comment:
If you can type the cache object it would be more type safe

Reply:
Where is it defined..the `cache` object?

Reply:
Ok, I guess `Object` should be fine then...

Reply:
If the `cache` can cache other things beside `modules` [this part](https://github.com/webpack/docs/wiki/internal-webpack-plugins#cacheplugincache) could be adapted..